### ISSUE TITLE
Can’t Install File Editor 5.8.0

### ISSUE THREAD

**Issue by wilwizard1979** [home-assistant/addons#3555]
### Describe the issue you are experiencing

I can’t install the File Editor Update 5.8.0.
This issue is active since some weeks, but I decided to wait for the new Core updates to see if the situation would change. But it is not the case.
I’ll attach the complete message I receive when trying to do the update but in short it says “Failed to Call Service Update/Install”

![IMG_3644](https://github.com/home-assistant/addons/assets/108979356/ad52501a-c15a-4bef-8eff-971988cd197f)

### What type of installation are you running?

Home Assistant OS

### Which operating system are you running on?

Home Assistant Operating System

### Which add-on are you reporting an issue with?

File Editor

### What is the version of the add-on?

5.7.0

### Steps to reproduce the issue

1.go to Devices & Services
2.click on the update available (File editor Update)
3.click install
...

## System Information

version | core-2024.4.2
-- | --
installation_type | Home Assistant OS
dev | false
hassio | true
docker | true
user | root
virtualenv | false
python_version | 3.12.2
os_name | Linux
os_version | 6.6.20-haos
arch | x86_64
timezone | Europe/Berlin
config_dir | /config

<details><summary>Home Assistant Community Store</summary>

GitHub API | ok
-- | --
GitHub Content | ok
GitHub Web | ok
GitHub API Calls Remaining | 4949
Installed Version | 1.34.0
Stage | running
Available Repositories | 1409
Downloaded Repositories | 8

</details>

<details><summary>Home Assistant Cloud</summary>

logged_in | true
-- | --
subscription_expiration | 13 November 2024 at 01:00
relayer_connected | true
relayer_region | eu-central-1
remote_enabled | true
remote_connected | true
alexa_enabled | true
google_enabled | false
remote_server | eu-central-1-4.ui.nabu.casa
certificate_status | ready
instance_id | (removed)
can_reach_cert_server | ok
can_reach_cloud_auth | ok
can_reach_cloud | ok

</details>

<details><summary>Home Assistant Supervisor</summary>

host_os | Home Assistant OS 12.1
-- | --
update_channel | stable
supervisor_version | supervisor-2024.04.0
agent_version | 1.6.0
docker_version | 24.0.7
disk_total | 30.8 GB
disk_used | 23.2 GB
healthy | true
supported | true
board | ova
supervisor_api | ok
version_api | ok
installed_addons | File editor (5.7.0), Studio Code Server (5.15.0), Samba share (12.3.1), Node-RED (17.0.11), TasmoAdmin (0.30.0), Terminal & SSH (9.10.0), Zigbee2MQTT (1.36.1-1), PSA Car Controller (v3.5.0), ESPHome (2024.3.2), Mosquitto broker (6.4.0), RTSP Simple Server Add-on (v0.17.6), eufy-security-ws (1.8.0-2), Advanced SSH & Web Terminal (17.2.0), eWeLink Smart Home (1.4.3), Home Assistant Google Drive Backup (0.112.1)

</details>

<details><summary>Dashboards</summary>

dashboards | 4
-- | --
resources | 10
views | 23
mode | storage

</details>

<details><summary>Recorder</summary>

oldest_recorder_run | 4 April 2024 at 05:36
-- | --
current_recorder_run | 10 April 2024 at 20:51
estimated_db_size | 218.67 MiB
database_engine | sqlite
database_version | 3.44.2

</details>

<details><summary>Sonoff</summary>

version | 3.6.0 (966334d)
-- | --
cloud_online | 3 / 7
local_online | 3 / 3

</details>

<details><summary>Spotify</summary>

api_endpoint_reachable | ok
-- | --

</details>

### Anything in the Supervisor logs that might be useful for us?

```txt
logger:%20homeassistant.components.websocket_api.http.connection%0ASource%3A%20components/websocket_api/commands.py:239%0Aintegration:%20Home%20Assistant%20WebSocket%20API%20(documentation,%20issues)%0AFirst%20occurred:%2021:13:24%20(2%20occurrences)%0ALast%20logged:%2021:15:23%0A%0A%5B139647210310592%5D%20Error%20updating%20File%20editor:%20Can't%20install%20homeassistant/amd64-addon-configurator:5.8.0:%20500%20Server%20Error%20for%20http+docker://localhost/v1.43/images/homeassistant/amd64-addon-configurator:5.8.0/json:%20Internal%20Server%20Error%20(%22readlink%20/mnt/data/docker/overlay2:%20invalid%20argument%22)%0A%5B139647210408640%5D%20Error%20updating%20File%20editor:%20Can't%20install%20homeassistant/amd64-addon-configurator:5.8.0:%20500%20Server%20Error%20for%20http+docker://localhost/v1.43/images/homeassistant/amd64-addon-configurator:5.8.0/json:%20Internal%20Server%20Error%20(%22readlink%20/mnt/data/docker/overlay2:%20invalid%20argument%22)%0ATraceback%20(most%20recent%20call%20last):%0A%20%20File%20%22/usr/src/homeassistant/homeassistant/components/hassio/update.py%22,%20line%20168,%20in%20async_install%0A%20%20%20%20await%20async_update_addon(self.hass,%20slug=self._addon_slug,%20backup=backup)%0A%20%20File%20%22/usr/src/homeassistant/homeassistant/components/hassio/handler.py%22,%20line%2059,%20in%20_wrapper%0A%20%20%20%20data%20=%20await%20funct(*argv,%20**kwargs)%0A%20%20%20%20%20%20%20%20%20%20%20%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%0A%20%20File%20%22/usr/src/homeassistant/homeassistant/components/hassio/handler.py%22,%20line%20136,%20in%20async_update_addon%0A%20%20%20%20return%20await%20hassio.send_command(%0A%20%20%20%20%20%20%20%20%20%20%20%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%0A%20%20File%20%22/usr/src/homeassistant/homeassistant/components/hassio/handler.py%22,%20line%20593,%20in%20send_command%0A%20%20%20%20raise%20HassioAPIError(error.get(ATTR_MESSAGE))%0Ahomeassistant.components.hassio.handler.HassioAPIError:%20Can't%20install%20homeassistant/amd64-addon-configurator:5.8.0:%20500%20Server%20Error%20for%20http+docker://localhost/v1.43/images/homeassistant/amd64-addon-configurator:5.8.0/json:%20Internal%20Server%20Error%20(%22readlink%20/mnt/data/docker/overlay2:%20invalid%20argument%22)%0A%0AThe%20above%20exception%20was%20the%20direct%20cause%20of%20the%20following%20exception:%0A%0ATraceback%20(most%20recent%20call%20last):%0A%20%20File%20%22/usr/src/homeassistant/homeassistant/components/websocket_api/commands.py%22,%20line%20239,%20in%20handle_call_service%0A%20%20%20%20response%20=%20await%20hass.services.async_call(%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%0A%20%20File%20%22/usr/src/homeassistant/homeassistant/core.py%22,%20line%202543,%20in%20async_call%0A%20%20%20%20response_data%20=%20await%20coro%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%0A%20%20File%20%22/usr/src/homeassistant/homeassistant/core.py%22,%20line%202580,%20in%20_execute_service%0A%20%20%20%20return%20await%20target(service_call)%0A%20%20%20%20%20%20%20%20%20%20%20%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%0A%20%20File%20%22/usr/src/homeassistant/homeassistant/helpers/service.py%22,%20line%20971,%20in%20entity_service_call%0A%20%20%20%20single_response%20=%20await%20_handle_entity_call(%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%0A%20%20File%20%22/usr/src/homeassistant/homeassistant/helpers/service.py%22,%20line%201043,%20in%20_handle_entity_call%0A%20%20%20%20result%20=%20await%20task%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%0A%20%20File%20%22/usr/src/homeassistant/homeassistant/components/update/__init__.py%22,%20line%20162,%20in%20async_install%0A%20%20%20%20await%20entity.async_install_with_progress(version,%20backup)%0A%20%20File%20%22/usr/src/homeassistant/homeassistant/components/update/__init__.py%22,%20line%20466,%20in%20async_install_with_progress%0A%20%20%20%20await%20self.async_install(version,%20backup)%0A%20%20File%20%22/usr/src/homeassistant/homeassistant/components/hassio/update.py%22,%20line%20170,%20in%20async_install%0A%20%20%20%20raise%20HomeAssistantError(f%22Error%20updating%20%7Bself.title%7D:%20%7Berr%7D%22)%20from%20err%0Ahomeassistant.exceptions.HomeAssistantError:%20Error%20updating%20File%20editor:%20Can't%20install%20homeassistant/amd64-addon-configurator:5.8.0:%20500%20Server%20Error%20for%20http+docker://localhost/v1.43/images/homeassistant/amd64-addon-configurator:5.8.0/json:%20Internal%20Server%20Error%20(%22readlink%20/mnt/data/docker/overlay2:%20invalid%20argument%22)%0A
```

### Anything in the add-on logs that might be useful for us?

```txt
No errors have been reported when selecting the Add-On file editor.
The add-on works normally.
```

### Additional information

Nothing else to add

**Comment by koolikoff** [comment]
+1

**Comment by agners** [comment]
This looks like some type of data corruption in the Docker storage backend used under the hood :cry: 

You can try `ha supervisor repair`, and see if that helps. Unfortunately, not all such Docker corruption can be resolved. If that doesn't help, taking a full backup and restoring on a new installation is your best option.

**Comment by chrisky** [comment]
+1.  I get "Failed to install add-on 'AddonManager.install' blocked from execution, no host internet connection".  There is internet conneciton.

**Comment by wilwizard1979** [comment]
I tried the repair before backup and restore in a new installation. That was the only one that worked for me. Thanks @agners.
If you are running in a VM, be aware of some few things:
-make sure that your hardware configuration is the same in both VMs
- be aware that for some reason the boot sequence in a new VM will put the “CD-ROM” as first in the boot sequence. This will let you fail to boot. There is a thread in Home Assistant forum that I posted how to fix it. Can’t just find easily now as my internet connection is poor where I am.
